cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +356,38 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubeadmVersion    string
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+
+	// Skip this check for "super-custom builds", where apimachinery/the overall codebase version is not set.
+	if kubever.KubeadmVersion == "v0.0.0" {
```

Comment:
`strings.HasPrefix(kubever.KubeadmVersion, "v0.0.0")`; it's actually a much longer version